### PR TITLE
Refactor and new functions supported

### PR DESCRIPTION
**Pull request by irykoon**
Improvement:
- Replace the tempfile with a variable 

Refactors:
- Move get_tor_start_command() to torrc-parser 
- Refactor code into a function extra_specified_torrc() 
- Remove redundant code 

New function:
value_of_disable_network() call shows the final value of DisableNetwork adopted by Tor.  `$TOR_ENABLED` is also used as part of the `tor_enabled_check` in the `whonixcheck`

### CODE REVIEW

**Comment by adrelanos** [comment]
Do you think usr/lib/anon-gw-anonymizer-config/torrc-parser would fit better into anon-shared-helper-scripts?

**Comment by irykoon** [comment]
Patrick Schleizer:
> Do you think usr/lib/anon-gw-anonymizer-config/torrc-parser would fit better into anon-shared-helper-scripts?
> 

I agreed. According to the README descriptions, it is a better fit into
anon-shared-helper-scripts.

However, I really wish we could have anon-gw-helper-scripts and
anon-ws-helper-scripts, therefore, we don't have to write a lot of
checking code like this:

```
   ## Skip this test, if not running on Whonix-Gateway.
   if [ ! -e "/usr/share/anon-gw-base-files/gateway" ]; then
      TOR_ENABLED="1"
      return 0
   fi
```

It seems anon-gw-base-files and anon-ws-base-files can then be merged
into anon-gw-helper-scripts and anon-ws-helper-scripts.

What do you think?

File: usr/bin/anon-verify [hunk 1]
```diff
@@ -181,6 +161,11 @@ $start_menu_instructions_system_first_part Terminal
 
 parse_cmd_options "$@"
 
+## many functions in this script is provided by:
```
Comment (adrelanos) [inline usr/bin/anon-verify]: Small nit: Please use the Whonix style for this.

    ## provides:  get_tor_start_command

etc.

(I haven't seen any non-Whonix general way to comment this, so invented that way.)

**Comment by adrelanos** [comment]
> I really wish we could have anon-gw-helper-scripts and anon-ws-helper-scripts, therefore, we don't have to write a lot of checking code like this:

Well, but then we would have lots of code duplication for other source code that is currently provided by anon-shared-helper-scripts?

**Comment by adrelanos** [comment]
At the moment we don't need torrc parsing in Whonix-Workstation. But in future it may be useful in context of https://www.whonix.org/wiki/Dev/anon-ws-disable-stacked-tor. So having torrc parsing code in a gateway/workstation (and even Whonix) agnostic package anon-shared-helper-scripts helps to make it very future proof.

**Comment by irykoon** [comment]
I agree it is better to use a uniform-looking comments on the function calls. Let's use this form:
```
## provides: a_function_call # note it is not a_function_call()
## sets: $a_variable
source /path/to/another/script
```

**Comment by irykoon** [comment]
Patrick Schleizer:
>> I really wish we could have anon-gw-helper-scripts and anon-ws-helper-scripts, therefore, we don't have to write a lot of checking code like this:
> 
> Well, but then we would have lots of code duplication for other source code that is currently provided by anon-shared-helper-scripts?
> 

I see. I mean we can still keep the anon-shared-helper-scripts package
as long as the code in it is useful to both VMs.

But if there's any code that we can guarantee that is exclusively useful
for one VM, one can get it into anon-gw-helper-scripts or
anon-ws-helper-scripts?

**Comment by adrelanos** [comment]
Sorry for late reply. Lost track of this one.

> But if there's any code that we can guarantee that is exclusively useful for one VM, one can get it into anon-gw-helper-scripts or anon-ws-helper-scripts?

I guess for simplicity and having fewer packages we'd still add to helper-scripts only.

----

Merge conflict. If still interested, please re-base and re-open or send a new pull request / open a new discussion.

**Pull request closed without merging**